### PR TITLE
feat: store appearance settings globally across accounts

### LINKED ISSUE #2144: Global Display and Language settings
#### Current behavior
- Display and Language settings are per account and can only be changed after account creation/import and login
- When Display and Language settings are changed, these changes are not visible on the login screen

#### Proposed behavior
- ~Account creation: Support 'default app mode' on Windows (and equivalent on Mac and Linux) and set appearance mode on first time launch to 'System' as default~ moved to #2177 
- Login screen: Add option to change global Display and Language setting on the login screen (https://i.imgur.com/FQZMMcG.png)
- Settings: Keep option to change global Display and Language setting in Settings post login
- Additional account creation: When a new account is created, the global setting is applied

**********
#### User stories
- User story 1
     - As a user _who opted to change the display setting,_
     - I want _this setting to apply to my most frequently visited screens_
     - Because _that's why I chose the setting_

- User story 2 --> addressed by #2177 
     - As a user _who is new to Status_
     - I want _for the application to respect my system Language and Display settings on launch (account creation/import)_
     - Because _these are settings I prefer for my applications_
     
## Notes
A 3rd user stories is not covered by this solution as it has conflicting requirements
- User story 3
     - As a user _who has multiple accounts,_ 
     - I want _to set a language for each_ 
     - So that _I can switch between languages_

I suspect changing these might be different issues:
- Listen to dark mode system setting on windows, mac, Linux
- Change toolbar to respect appearance setting
- Hide Language on login screen when only one language (EN) is available

### PR DESCRIPTION
**Pull request by PascalPrecht**
As discussed in https://github.com/status-im/status-desktop/issues/2144#issuecomment-817791172 and https://github.com/status-im/status-desktop/discussions/2145, it's no longer desired
to have different appearance settings across multiple accounts.
Instead, the appearance setting should apply globally to all accounts,
essentially bypassing the individual setting stored in status-go.

This commit introduces a new global setting called `theme` which,
at the time of introducing this commit, can be either:

0 => Light
1 => Dark
2 => System

Because those enum values matches the `AppearanceContainer.Theme` enum,
this commit removes it completely and simply relies on QML's built-in
`Universal.[Light|Dark|System] variants respectively.

Closes #2144

### CODE REVIEW

File: ui/app/AppLayouts/Profile/Sections/AppearanceContainer.qml [hunk 1]
```diff
@@ -286,22 +280,13 @@ ScrollView {
...
             }
-
-            // For the case where the theme was finally loaded by status-go in init(),
-            // update the theme in qml
-            Connections {
-                target: profileModel
-                onProfileChanged: {
-                    root.updateTheme(profileModel.profile.appearance)
-                }
-            }
```
Comment (PascalPrecht) [inline ui/app/AppLayouts/Profile/Sections/AppearanceContainer.qml:304]: We're no longer interested in the setting coming from status-go, so we can get rid off this connection here entirely

File: ui/app/AppMain.qml [hunk 2]
```diff
@@ -150,13 +150,6 @@ RowLayout {
         property bool compatibilityMode: true
     }
 
-    Settings {
-        id: globalSettings
-        category: "global"
-        fileName: profileModel.globalSettingsFile
-        property string locale: "en"
-    }
-
```
Comment (PascalPrecht) [inline ui/app/AppMain.qml:159]: This is moved to `ui/main.qml` because we need `globalSettings` there

File: ui/imports/Style.qml [hunk 3]
```diff
@@ -10,11 +11,10 @@ QtObject {
 
     property var changeTheme: function (theme) {
         switch (theme) {
-            case "light": current = lightTheme; break;
-            case "dark": current = darkTheme; break;
+            case Universal.Light: current = lightTheme; break;
+            case Universal.Dark: current = darkTheme; break;
             default: current = lightTheme; console.log('Unknown theme. Valid themes are "light" and "dark"')
```
Comment (PascalPrecht) [inline ui/imports/Style.qml:16]: Something very important to notice here:

There has never been an option to actually: Set the theme to dark, or light, if system is dark or light respectively.
While we do have enum variants `Dark`, `Light` and `System` that can be used to tell our app what theme to use, there doesn't seem to be a way to actually determine **what** `System` is.

Therefore, so far we've always defaulted to `Light`.

In other words, there's no such thing as "default to `System`". I assume there's a way to determine what the current configured theme on the OS level is inside Qt/Nim, but there's certainly no way in QML and I couldn't find anything about it either.

Comment (jrainville) [reply ui/imports/Style.qml:16]: Aaaah, and I thought it was a Windows bug. So it actually never worked even on Mac 🤦

**Review by jrainville — APPROVED** [review]

**Comment by anastasiyaig** [comment]
@PascalPrecht can we please rebase so Mac build is fixed and created?

**Comment by PascalPrecht** [comment]
@anastasiyaig this is rebased now

**Comment by anastasiyaig** [comment]
@PascalPrecht 

i've tested different combinations of language and theme for new and imported accounts, looks good - once applied , a setting is present to all accounts in application

Let's merge it!
